modules/apps/portal/portal-messaging/src/main/java/com/liferay/portal/messaging/internal/DefaultMessageBus.java

```diff
@@ -201,120 +214,32 @@ public void removedService(
 					"DestinationWorkerConfiguration"
 			).build());
 
-		_serviceTrackerList = ServiceTrackerListFactory.open(
-			bundleContext, MessageBusInterceptor.class);
+		_messageBusInterceptorServiceTrackerList =
+			ServiceTrackerListFactory.open(
+				bundleContext, MessageBusInterceptor.class);
 	}
 
 	@Deactivate
 	protected void deactivate() {
-		_serviceTrackerList.close();
+		_messageBusInterceptorServiceTrackerList.close();
 
 		_serviceRegistration.unregister();
 
 		_messageListenerServiceTracker.close();
 
-		_messageBusEventListeners.clear();
-
 		shutdown(true);
 
+		_destinationEventListenerServiceTracker.close();
+
 		for (Destination destination : _destinations.values()) {
 			destination.destroy();
 		}
 
 		_destinations.clear();
```

Comment:
@W2jx Is this part still needed? When you close a `ServiceTracker`, all the tracked ones will be untracked, during which `removedService` will be called for each.